runtime/runtime/src/actions.rs

```diff
@@ -175,34 +182,42 @@ pub(crate) fn action_function_call(
         promise_results,
     );
     if let Some(err) = err {
-        if let VMError::StorageError(storage) = err {
-            let err: StorageError =
-                borsh::BorshDeserialize::try_from_slice(&storage).expect("Borsh cannot fail");
-            return Err(err);
-        }
-        result.result = Err(ActionErrorKind::FunctionCall(err).into());
-        if let Some(outcome) = outcome {
-            result.gas_burnt += outcome.burnt_gas;
-            result.gas_burnt_for_function_call += outcome.burnt_gas;
-            // Runtime in `generate_refund_receipts` takes care of using proper value for refunds.
-            // It uses `gas_used` for success and `gas_burnt` for failures. So it's not an issue to
-            // return a real `gas_used` instead of the `gas_burnt` into `ActionResult` for
-            // `FunctionCall`s error.
-            result.gas_used += outcome.used_gas;
-            result.logs.extend(outcome.logs.into_iter());
+        match err {
+            VMError::FunctionCallError(err) => {
+                result.result = Err(ActionErrorKind::FunctionCallError(err).into());
+                if let Some(outcome) = outcome {
+                    result.gas_burnt += outcome.burnt_gas;
+                    result.gas_burnt_for_function_call += outcome.burnt_gas;
+                    // Runtime in `generate_refund_receipts` takes care of using proper value for refunds.
+                    // It uses `gas_used` for success and `gas_burnt` for failures. So it's not an issue to
+                    // return a real `gas_used` instead of the `gas_burnt` into `ActionResult` for
+                    // `FunctionCall`s error.
+                    result.gas_used += outcome.used_gas;
```

Comment:
I think we should leave comments about overflow that is not expected to happen at every occasion, so it is easier to review math operations in the future.

Reply:
Actually wrapped math here in case we would ever hit this limit.